### PR TITLE
Add TypeScript types

### PR DESCRIPTION
**Pull request by charlesfries**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    };
+    Return: void;
+  }> {}
+}
+
+declare module 'ember-breadcrumb-trail/helpers/breadcrumbs' {
+  import Helper from '@ember/component/helper';
+
+  export default class Breadcrumbs extends Helper<{
+    Return: { title: string; data: { route: string; model: string } }[];
```
Comment (Windvis) [inline index.d.ts]: The data object is everything that's passed as a named argument into the `{{breadcrumb}}` helper, so the route and model keys aren't guaranteed.

Should it be something like this maybe?
```suggestion
    Return: { title: string; data: Record<string, unknown> }[];
```

**Review by Windvis — COMMENTED** [review]
Hey, thanks for the PR!

I'm fairly new to to TS and how this works in addons. What's the benefit of these types? How will apps use this? Is publishing the file enough, or does this need some package.json tweaks to let TS find the types?

(I'll see if I can find some time in the near future to revive this addon and convert it to a v2 addon with full Glint support. I'm not really using it myself and I wasn't aware anyone else was either 😅)

**Comment by Windvis** [comment]
CI faillures seem to be caused by the very outdated node 12 this repo is still using 😄. We'll just ignore those for now.

File: index.d.ts (hunk 1 above)
Comment (charlesfries) [reply index.d.ts]: Yes you are correct--this was an oversight on my end.

**Comment by charlesfries** [comment]
@Windvis I am nearly positive that just adding the `*.d.ts` files will make the types available to consuming applications (take a look at this PR for `ember-data-model-fragments` as an example of a JS addon shipping types: https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/443)

**Comment by Windvis** [comment]
> @Windvis I am nearly positive that just adding the *.d.ts files will make the types available to consuming applications (take a look at this PR for ember-data-model-fragments as an example of a JS addon shipping types: https://github.com/adopted-ember-addons/ember-data-model-fragments/pull/443)

But you still need to import that index.d.ts file in your app I assume? Something like `import 'ember-breadcrumb-trail';`? Or is that not how it works? You would also need to add the helpers to your glint registry manually, no? Since this only exposes the base types of the helper? Are the types in the current PR useful without Glint?

Sorry, many questions, but I don't use TS in apps yet, so I've little experience with it 😄.

**Comment by Windvis** [comment]
I think it might also be possible to expose types from a declarations folder and matching the addon folder structure so TS can find them without having to import an index file first. Something like: `declarations/helpers/breadcrumb.d.ts` maybe?

Anyway, I'll try to take a more detailed look at it soon. Thanks for the nudge!

**Comment by Windvis** [comment]
@charlesfries I've finally finished the TS conversion PR + released it as 1.0.0. Sorry for the delay :smile:

**Pull request closed without merging**